cvise/utils/testing.py

```diff
@@ -830,8 +848,8 @@ def maybe_schedule_job(self, pool: pebble.ProcessPool) -> bool:
                 self.schedule_fold(pool, folding_state)
                 return True
         # 3. Attempting a transformation using the next heuristic in the round-robin fashion.
-        if any(ctx.state is not None for ctx in self.pass_contexts):
-            while self.pass_contexts[self.next_pass_id].state is None:
+        if any(ctx.stage == PassStage.ENUMERATING and ctx.state is not None for ctx in self.pass_contexts):
+            while self.pass_contexts[self.next_pass_id].state is None or self.pass_contexts[self.next_pass_id].stage != PassStage.ENUMERATING:
```

Comment:
Note: This is hardly readable, but the refactoring in #257 will make it much more concise.